ddtrace/internal/utils/cache.py

```diff
@@ -107,10 +113,22 @@ def cached_wrapper(f):
 def callonce(f):
     # type: (Callable[[], Any]) -> Callable[[], Any]
     """Decorator for executing a function only the first time."""
-
-    argspec = getargspec(f)
-    if argspec.args or argspec.varargs or argspec.keywords or argspec.defaults or isgeneratorfunction(f):
-        raise ValueError("The callonce decorator can only be applied to functions with no arguments")
+    if PY2:
+        argspec = getargspec(f)
+        if argspec.args or argspec.varargs or argspec.keywords or argspec.defaults or isgeneratorfunction(f):
+            raise ValueError("The callonce decorator can only be applied to functions with no arguments")
+    else:
+        argspec = getfullargspec(f)
```

Comment:
are we able to have a backported/compatible version of `getfullargspec` in `ddtrace/internal/compat.py` ?

Reply:
@brettlangdon Is a backport necessary? This should only be a problem for Python 3.11 onwards.

Reply:
Moved this to `ddtrace/internal/compat.py`.